compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -491,8 +515,12 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
         case EnumTag   =>
           val sym       = const.symbolValue
           val ownerName = internalName(sym.owner)
-          val fieldName = sym.javaSimpleName.toString
-          val fieldDesc = toTypeKind(sym.tpe.underlying).descriptor
+          val fieldName = sym.name.mangledString.toString
+          val underlying = sym.info match {
+            case t: TypeProxy => t.underlying
+            case t => t
+          }
```

Comment:
Is this actually necessary? Could it be replaced by a call to widen?

Reply:
Added TODO comment